gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -352,136 +341,75 @@ public void testMergeFeature() {
-    Assert.assertEquals(layer.getItemLabel(), newFeature.getTitle());
-    Assert.assertEquals(layer, newFeature.getLayer());
+    Assert.assertEquals(TEST_LAYER.getItemLabel(), newFeature.getTitle());
+    Assert.assertEquals(TEST_LAYER, newFeature.getLayer());
     Assert.assertNull(newFeature.getCustomId());
     Assert.assertNull(newFeature.getCaption());
     Assert.assertEquals(point, newFeature.getPoint());
-    Assert.assertEquals(user, newFeature.getCreated().getUser());
-    Assert.assertEquals(user, newFeature.getLastModified().getUser());
+    Assert.assertEquals(TEST_USER, newFeature.getCreated().getUser());
+    Assert.assertEquals(TEST_USER, newFeature.getLastModified().getUser());
   }
 
   @Test
   public void testApplyAndEnqueue_observationMutation() {
-    User user = createTestUser();
-    localDataStore.insertOrUpdateUser(user).test().assertComplete();
-
-    Project project = createTestProject();
-    localDataStore.insertOrUpdateProject(project).test().assertComplete();
-
-    Layer layer = project.getLayers().get(0);
-    FeatureMutation featureMutation =
-        createTestFeatureMutation(user.getId(), project.getId(), layer.getId());
-    localDataStore.applyAndEnqueue(featureMutation).test().assertComplete();
-
-    ObservationMutation mutation =
-        createTestObservationMutation(
-            project.getId(),
-            featureMutation.getFeatureId(),
-            layer.getId(),
-            layer.getForm().get().getId(),
-            user.getId());
+    localDataStore.insertOrUpdateUser(TEST_USER).subscribe();
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).subscribe();
+    localDataStore.applyAndEnqueue(TEST_FEATURE_MUTATION).subscribe();
+
+    ObservationMutation mutation = TEST_OBSERVATION_MUTATION;
     localDataStore.applyAndEnqueue(mutation).test().assertComplete();
 
     ImmutableList<Mutation> savedMutations =
         localDataStore.getPendingMutations(mutation.getFeatureId()).blockingGet();
     Assert.assertEquals(2, savedMutations.size());
-
     // ignoring the first item, which is a FeatureMutation. Already tested separately.
-    ObservationMutation savedMutation = (ObservationMutation) savedMutations.get(1);
-    Assert.assertEquals(mutation.getResponseDeltas(), savedMutation.getResponseDeltas());
-    Assert.assertEquals(mutation.getType(), savedMutation.getType());
-    Assert.assertEquals(mutation.getUserId(), savedMutation.getUserId());
-    Assert.assertEquals(mutation.getProjectId(), savedMutation.getProjectId());
-    Assert.assertEquals(mutation.getFeatureId(), savedMutation.getFeatureId());
-    Assert.assertEquals(layer.getId(), savedMutation.getLayerId());
-    Assert.assertEquals(mutation.getClientTimestamp(), savedMutation.getClientTimestamp());
-    Assert.assertEquals(0, savedMutation.getRetryCount());
-    Assert.assertNull(savedMutation.getLastError());
+    assertObservationMutation(mutation, (ObservationMutation) savedMutations.get(1));
```

Comment:
You probably will also want to `assertThat(mutation).isInstanceOf(ObservationMutation.class)`